app/models/social_networking/like.rb

```diff
@@ -8,15 +8,54 @@ class Like < ActiveRecord::Base
     # a Participant may only Like an item once
     validates :item_id, uniqueness: { scope: [:item_type, :participant_id] }
 
-    scope :for_today, lambda {
-      where(arel_table[:created_at]
-              .gteq(Date.today.beginning_of_day)
-              .and(arel_table[:created_at].lteq(Date.today.end_of_day)))
-    }
+    scope :for_today,
+          lambda {
+            where(arel_table[:created_at]
+                    .gteq(Date.today.beginning_of_day)
+                    .and(arel_table[:created_at].lteq(Date.today.end_of_day)))
+          }
 
-    scope :for_week, lambda {
-      where(arel_table[:created_at]
-              .gteq(Time.current.advance(days: -7).beginning_of_day))
-    }
+    scope :for_week,
+          lambda {
+            where(arel_table[:created_at]
+                    .gteq(Time.current.advance(days: -7).beginning_of_day))
+          }
+
+    def item_description
```

Comment:
answers/questions:

* The decorator/presenter pattern seemed like a reasonable choice to me. What advantages would static methods have? Possibly a memory use improvement?
* I'm not sure I understand the second question. Can you give an example?
* This may be a case of premature abstraction, but one reason would be that it achieves separation of concerns: the SharedItem class is tied to the database and its primary function is arguably to port data between the database and Ruby. Shareable is a focused class with a specific function. It is more easily testable.
* The class already exists: https://github.com/cbitstech/social_networking/blob/master/app/models/social_networking/shareable.rb